docs/rules/no-constructor-return.md

```diff
@@ -0,0 +1,52 @@
+# Disallow returning value in constructor (no-constructor-return)
+
+In JavaScript returning value in constructor is allowed, however this may be a mistake. Forbidding this pattern prevents mistake resulting from unfamiliarity with the language or copy-paste error.
+
+Note that returning nothing with flow control is allowed.
```

Comment:
Should this paragraph be in the "Rule Details" section?

This first section should just be about general background/motivation for the rule. However, I think I'm reading "allowed" in this paragraph to mean, the rule will allow it/not warn (which is not the same as what "allowed" means in the previous paragraph-- JavaScript spec/engines allow returning a value; that paragraph is general background information).